Modules/_sqlite/connection.c

```diff
@@ -1139,6 +1139,20 @@ destructor_callback(void *ctx)
     }
 }
 
+static int
+check_num_params(pysqlite_Connection *self, const int n, const char *name)
+{
+    int limit = sqlite3_limit(self->db, SQLITE_LIMIT_FUNCTION_ARG, -1);
+    assert(limit >= 0);
```

Comment:
As long as we call this on a valid database pointer (checked via `pysqlite_check_connection`), we can assert we won't fail here.